test/unittest/tests/test_prepare_mod_sort.cpp

```diff
@@ -0,0 +1,373 @@
+{
+    ConnectionWrapper conn(DB_HOME);
+    WT_SESSION_IMPL *session = conn.createSession();
+
+    WT_BTREE btrees[2];
+    WT_TXN_OP ops[10];
+    int key_sz = 6;
+    bool ret;
+    WT_ITEM *keys[key_sz];
+
+    allocate_key_space(session, key_sz, keys);
+
+    for (int i = 0; i < 6; i++)
+        init_key(keys[i], random_keys(3));
+
+    init_btree(&btrees[0], BTREE_COL_VAR, 1);
+    init_btree(&btrees[1], BTREE_ROW, 2);
+
+    // Column operations.
+    init_op(&ops[0], &btrees[0], WT_TXN_OP_BASIC_COL, 12, NULL);
+    init_op(&ops[1], &btrees[0], WT_TXN_OP_BASIC_COL, 45, NULL);
+
+    // Row operations.
+    init_op(&ops[2], &btrees[1], WT_TXN_OP_REF_DELETE, WT_RECNO_OOB, keys[0]);
+    init_op(&ops[3], &btrees[1], WT_TXN_OP_BASIC_ROW, WT_RECNO_OOB, keys[1]);
+    init_op(&ops[4], &btrees[1], WT_TXN_OP_BASIC_ROW, WT_RECNO_OOB, keys[2]);
+    init_op(&ops[5], &btrees[1], WT_TXN_OP_BASIC_ROW, WT_RECNO_OOB, keys[3]);
+    init_op(&ops[6], &btrees[1], WT_TXN_OP_BASIC_ROW, WT_RECNO_OOB, keys[4]);
+    init_op(&ops[7], &btrees[1], WT_TXN_OP_BASIC_ROW, WT_RECNO_OOB, keys[5]);
+
+    // Non key'd operations.
+    init_op(&ops[8], &btrees[0], WT_TXN_OP_TRUNCATE_COL, WT_RECNO_OOB, NULL);
+    init_op(&ops[9], &btrees[1], WT_TXN_OP_REF_DELETE, WT_RECNO_OOB, NULL);
+
+    __wt_qsort_r(&ops, 10, sizeof(WT_TXN_OP), __ut_txn_mod_compare, NULL);
+    ret = __mod_ops_sorted(ops, 10);
+
+    // Free the allocated scratch buffers.
+    for (int i = 0; i <= key_sz; i++)
+        __wt_scr_free(session, &keys[i]);
+
+    REQUIRE(ret == true);
+}
+
+// Test sorting by b-tree ID. All operations have the same key.
+TEST_CASE("B-tree ID sort test", "[mod_compare]")
+{
+    ConnectionWrapper conn(DB_HOME);
+    WT_SESSION_IMPL *session = conn.createSession();
+
+    WT_BTREE btrees[6];
+    WT_TXN_OP ops[6];
+    int key_sz = 1;
+    bool ret;
+    WT_ITEM *keys[key_sz];
+
+    allocate_key_space(session, key_sz, keys);
+
+    init_key(keys[0], "1");
+
+    for (int i = 0; i < 6; i++)
+        init_btree(&btrees[i], BTREE_ROW, rand() % 400);
+
+    for (int i = 0; i < 6; i++)
+        init_op(&ops[i], &btrees[i], WT_TXN_OP_BASIC_ROW, WT_RECNO_OOB, keys[0]);
+
+    __wt_qsort_r(&ops, 6, sizeof(WT_TXN_OP), __ut_txn_mod_compare, NULL);
+    ret = __mod_ops_sorted(ops, 6);
+
+    // Free the allocated scratch buffers first.
+    for (int i = 0; i <= key_sz; i++)
+        __wt_scr_free(session, &keys[i]);
+
+    REQUIRE(ret == true);
+}
+
+// Test sorting by keyedness, key'd operations all have the same key and recno.
+TEST_CASE("Keyedness sort test", "[mod_compare]")
+{
+    ConnectionWrapper conn(DB_HOME);
+    WT_SESSION_IMPL *session = conn.createSession();
+
+    WT_BTREE btrees[12];
+    WT_TXN_OP ops[12];
+    int key_sz = 1;
+    bool ret;
+    WT_ITEM *keys[key_sz];
+
+    allocate_key_space(session, key_sz, keys);
+
+    init_key(keys[0], "1");
+
+    for (int i = 0; i < 6; i++)
+        init_btree(&btrees[i], BTREE_ROW, rand() % 100);
```

Comment:
Is there any issue with picking the same btree id for two btrees? I expect this is ok

Reply:
You're right, that should be fine.